### ISSUE TITLE
Recalibrate M3 default priors: equal-means softmax + matched simple; document that c vs a comparisons need softmax

### ISSUE THREAD

**Issue by GidonFrischkorn** [venpopov/bmm#364]
## Summary

The M3 default priors are poorly calibrated for naive use — surfaced by reviewer comments on the M3 tutorial manuscript. Two concrete failure modes:

1. **Cross-rule prior-predictive discrepancy.** The `simple` and `softmax` choice rules ship with very different default priors, so they imply very different prior-predictive performance. For the canonical design `n = (correct 1, other 2, npl 3)` the prior means give **p(correct) ≈ 0.55** (`simple`) vs **≈ 0.89** (`softmax`). Marginal likelihoods are highly prior-sensitive, so users who keep the defaults and compare models via **bridge sampling / Bayes factors** get conclusions driven by the miscalibrated prior, not by the data.

2. **Asymmetric `a`/`c` intercept means break a fair `c = a` test under cell-means coding.** `a` (general activation) and `c` (context activation) have different `main` means (`c` > `a`). Under **cell-means coding** (`c ~ 0 + condition`, `a ~ 0 + condition`) — often the more accessible parameterization for measurement-model analysis — the `main` prior is applied to every cell, so the implied prior on `c − a` is **not centered at 0**, biasing any test of "context and general activation are equal."

We recommend custom priors for hypothesis testing, but many users rely on the defaults, so the defaults must be reasonable.

## Current defaults

```
ss  simple   a: N(1,   0.5)   c: N(1.5, 0.5)                       effects N(0, 0.5)
ss  softmax  a: N(2,   1)     c: N(3,   1)                  a-eff N(0, 0.5)   c-eff N(0, 2)
cs  simple   a: N(1,   0.5)   c: N(1.5, 0.5)   f: logistic(0,1)    effects N(0, 0.5)/N(0, 1)
cs  softmax  a: N(3,   1)     c: N(3,   1)     f: logistic(0,1)    a-eff N(0,0.5) c-eff N(0,2) f-eff N(0,1)
custom fallback (simple):   log N(1,1)   identity N(10,4)   logit logistic(0,1)/N(0,0.5)
custom fallback (softmax):  log N(0,1)   identity N(1,1)    logit logistic(0,1)/N(0,0.5)
```

Problems: (i) `a` and `c` means differ within every rule; (ii) the two rules sit at very different prior-predictive levels; (iii) the `softmax` `c` effects prior `N(0, 2)` is ~4.5× wider than `simple`'s `N(0, 0.5)` with no justification (on both log and identity scales a 1-SD shift multiplies activation by `exp(σ)`, so the scales are equivalent and the effect priors should match); (iv) the custom-version fallback repeats the same per-rule asymmetry.

## What prior-predictive simulation shows

Drawing `a`, `c` (and `f`) from candidate priors and mapping them through the M3 activation→probability rule (parity-checked against the package internal, exact to machine precision) yields the design decision below.

- **The rules behave very differently because `simple` normalizes raw activations while `softmax` exponentiates them.** Under `softmax`, even a modest `c − a` gap produces a large probability ratio; under `simple`, the correct category can only dominate when `c` is *substantially* larger than `a`.
- **softmax** — equal means `μ_a = μ_c` **can** span a broad p(correct) range. This is the chosen default: it centers `c − a` at 0 (fair `c = a` test) **and** stays broad/uninformative.
- **simple** — equal means is **structurally capped** below the empirical range: at the prior center the ceiling is `2 / (2 + n_other)` (≈ 0.33 for `n_other = 4`, 0.50 for `n_other = 2`), never reaching ~0.95. So the `simple` priors are instead **calibrated to reproduce the `softmax` prior predictive**.

### Chosen calibration (decision: equal-means softmax; matched simple)

Across representative designs, the `simple` rule reproduces the `softmax` equal-means prior predictive when **`c − a ≈ 3` on the log scale** — and this required gap is robust to the chosen center. Recommended pairing:

| rule | prior (`a`, `c`) | `c − a` center | real-design (`1,4,10`) p(correct) 5–95% |
|------|------------------|----------------|------------------------------------------|
| `softmax` (identity) | `a, c ~ N(3, 1)` (equal means) | **0** | ≈ [0.44, 0.96] |
| `simple` (log) | `a ~ N(0, 1)`, `c ~ N(3, 1)` | 3 | ≈ [0.40, 0.96] |

Quantile agreement between the two rules is within ≤ 0.06 across the `(1,4,10)` and `(1,2,3)` designs (≤ 0.02 over the upper half). A flatter, more-uninformative alternative is `softmax a,c ~ N(2.5, 1)` ↔ `simple a ~ N(-1, 1), c ~ N(2, 1)` (median ≈ 0.70 vs 0.81); same `c − a` gap of 3.

### Effect priors — evaluated, kept at `N(0, 0.5)`

Since the intercept→probability map is nonlinear, the new intercepts could in principle make the `N(0, 0.5)` effect prior too restrictive or too liberal. Simulating the implied between-condition Δp(correct) shows they do **not**:

- **`c`**: implied 5–95% effect range ≈ ±0.13–0.16 (≈ ±15 pp), essentially unchanged by the intercept move (softmax sd 0.087→0.082; simple ≈ 0.073). A sensible, weakly-informative range for condition/covariate effects.
- **`a`**: smaller p-impact (sd 0.034→0.018 under softmax) because `a` adds to *both* the correct and "other" categories, so it has low leverage on p; this is structural, and the link-scale prior stays consistent with `c` (both scale activation by `exp(effect)`). Appropriate as-is.

So the effect priors are kept at `N(0, 0.5)`. The only adjustment is harmonizing the mis-scaled **`softmax` `c`-effect `N(0, 2)` → `N(0, 0.5)`**: simulation confirms `N(0, 2)` lets a *single* condition effect span nearly the entire 0–1 probability range — far too liberal, and the source of part of problem 1.

## Proposed fix

- **softmax (identity link):** `a` and `c` share `main` mean & SD → `c − a` centered at 0. `a, c ~ N(3, 1)`.
- **simple (log link):** `a ~ N(0, 1)`, `c ~ N(3, 1)` — calibrated so the prior predictive matches `softmax`, restoring cross-rule comparability (problem 1).
- **Effects:** set the `a`/`c` `effects` priors to one shared SD `N(0, 0.5)` (drops the `N(0, 2)` outlier).
- **`f` (filtering):** `logistic(0, 1)` on the logit scale is already uniform(0, 1) on the probability scale — keep it; harmonize the `f` effects SD across rules and the custom fallback.
- **Custom-version fallback** (`check_model.m3_custom()`): align the per-rule `log`/`identity` centers with the calibrated ss/cs values so a custom parameter gets a prior-predictively comparable default regardless of rule.
- Leave the fixed scaling parameter `b` (`0.1` simple / `0` softmax) **unchanged** (scaling identification constraint).

The exact numbers should be re-confirmed by prior-predictive simulation against the package's representative example design before merging.

## Documentation: `simple` is not suited for comparing `c` vs `a`

A direct consequence of the calibration: because the `simple`-rule default **must** encode `c ≫ a` (a `c − a` gap of ~3 on the log scale) to reach realistic accuracy — and because a symmetric `c = a` prior cannot reach that range under `simple` — **the `simple` rule is not well suited for directly comparing or testing context (`c`) against general (`a`) activation.** The `softmax` rule, whose equal-means default centers `c − a` at 0, supports this comparison directly.

This should be documented in the M3 vignette and `?m3`: when the research question concerns the `c` vs `a` contrast (or any direct comparison of activation sources), recommend the `softmax` choice rule; note that `simple`-rule defaults are asymmetric by construction.

## Acceptance criteria

- [ ] `softmax` uses equal `a`/`c` means (`c − a` centered at 0); `simple` priors reproduce the `softmax` prior predictive (quantile abs-diff ≤ ~0.05 on representative designs).
- [ ] Prior-predictive p(correct) spans ~[0.40, 0.95] (central ~90%) for **both** rules.
- [ ] `a`/`c` `effects` priors share one SD; custom fallback harmonized; `b` fixed values unchanged.
- [ ] Vignette + `?m3` document that direct `c` vs `a` comparisons should use `softmax`, and that `simple` defaults are asymmetric by construction.
- [ ] Tests and `NEWS.md` updated.

## Notes

- Default priors can only be calibrated for a *representative* design; the implied performance still varies with each user's `num_options` / category structure — this should be documented.
- Pairs well with #194 (a `plot` method for `default_priors`), which would let users visually check the implied prior predictive for their own design.
- Files: `.m3_version_table` and `check_model.m3_custom()` in `R/model_m3.R`.
